### PR TITLE
Examples and README

### PR DESCRIPTION
**Pull request by jangolle**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,284 @@
...
+    }
+);
+```
+
+2. Add listener on `worker-exit` event of `Master`
+
+```javascript
+master.on('worker-exit', async (reason, workerId) => {
+    switch (reason.constructor) {
+        case ExitReasons.ExternalSignal:
```
Comment (WoZ) [inline README.md]: I hadn't found any mention that reason is an instance of `ExitReason.*` class. Up for me it's better to use `if (reason instanceof ExitReasons.ExternalSignal) {...} elseif () {...}`. It's more obvious than having `reason.constructor` that may be a regular constant field with constants like `ExitReasons.ExternalSignal`.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,284 @@
...
+        case ExitReasons.AbnormalExit:
+            // restart worker if something abnormal happened or external process killed worker by signal
+            await master.restartWorkerById(workerId);
+            break;
+        default:
+            // for different cases just hard quit all app
+            const { code, signal } = reason.payload;
+            await quit(code, signal, true);
```
Comment (WoZ) [inline README.md]: What is `quit`? If you write code with that below just write a comment in this line that `quit` will be described later.

**Review by WoZ — CHANGES_REQUESTED** [review]
Great job!

I have a few comments to improve docs.
1. Suppose it's a good idea to include info about the difference with PM2: MIT license instead of GPL (check the license plz), more flexible programmatic control. As I see, PM2's programmatic control is just API to pm2 process and you may use only a given subset of strategies, like restart after N queries. With drover you may add any logic you want. PM2 uses "let it fail" concept while drover shares idea of control. E.g. difference about normal and abnormal stop. Feature to run different groups and manage them.
2. It's better to add a section with the code right after install block. Just like starting a very simple app. The idea to exploit user's attention (max 10-15 sec) to understand what we are talking about. User must come around that it's easy to use.
3. Maybe we need to add a separate section with a description of differences between PM2 and drover.

File: README.md (hunk 1 above)
Comment (jangolle) [reply README.md]: Agree, done!

File: README.md (hunk 2 above)
Comment (jangolle) [reply README.md]: Yes for sure, missed it

**Review by WoZ — APPROVED** [review]

**Pull request merged**